src/hookable.ts

```diff
@@ -113,15 +117,29 @@ export class Hookable <
   }
 
   callHook<NameT extends HookNameT> (name: NameT, ...args: Parameters<InferCallback<HooksT, NameT>>) {
-    return serialCaller(this._hooks[name] || [], args)
+    return this.callHookWith(serialCaller, name, ...args)
   }
 
   callHookParallel<NameT extends HookNameT> (name: NameT, ...args: Parameters<InferCallback<HooksT, NameT>>) {
-    return parallelCaller(this._hooks[name] || [], args)
+    return this.callHookWith(parallelCaller, name, ...args)
   }
 
   callHookWith<NameT extends HookNameT, CallFunction extends (hooks: HookCallback[], args: Parameters<InferCallback<HooksT, NameT>>) => any> (caller: CallFunction, name: NameT, ...args: Parameters<InferCallback<HooksT, NameT>>): void | ReturnType<CallFunction> {
-    return caller(this._hooks[name] || [], args)
+    parallelCaller(this._before || [], [name, ...args])
```

Comment:
Since _before, _after are not supposed to be async, we can probably use a (sync) loop util that calls them in same tick

Reply:
Another idea: We can create `context = {}` and pass it to before, after hooks giving them a way to preserve state